build.gradle

```diff
@@ -50,8 +55,8 @@ mavenPublishing {
     }
 }
 
-sourceCompatibility = 11
-targetCompatibility = 11
+sourceCompatibility = 17
```

Comment:
probably we should add in the readme that the lib uses java 17

Reply:
* On further thought, it is better not to bump up source/target compatibility, since no jdk 17 specific feature is used in the lib. It would only prevent applications written in jdk versions < 17 from using the lib.